### ISSUE TITLE
links in Multi lobby titles

### ISSUE THREAD

**Issue by Ocklepod** [ppy/osu-stable-issues#975]
### Type

Game behaviour

### Bug description

When inviting to a multiplayer lobby, and the title includes a link, clicking it will open the link instead of joining the lobby. For example "[Tournament name]: (link) vs Team B", when clicking somewhere else, e.g. "Team B", it will let you join, but if the lobby is just "(link)", it's not possible to join. I suggest making external links unclickable if inside an internal link.

### Version

b20220424

### Logs

[runtime.log](https://github.com/ppy/osu/files/8599782/runtime.log)
[session.log](https://github.com/ppy/osu/files/8599783/session.log)
[update_success.log](https://github.com/ppy/osu/files/8599784/update_success.log)

**Comment by bdach** [comment]
As far as I can tell (from the version number and the logs), this issue pertains to stable. This repository only accepts reports for lazer.

Reports for stable go over at https://github.com/ppy/osu-stable-issues, but only major game-breaking bugs have a good chance of being fixed, and this doesn't seem to be one.

**Comment by peppy** [comment]
@Ocklepod are you able to provide a screenshot of this?

**Comment by hburn7** [comment]
> @Ocklepod are you able to provide a screenshot of this?

In the screenshots here you can see that when hovering over the link text, it redirects the user to that link instead of the multiplayer lobby. Also for some reason the last parentheses is not hyperlinked. 

https://imgur.com/a/xkv8uqq

**Comment by peppy** [comment]
![Parallels Desktop 2022-05-18 at 05 55 46](https://user-images.githubusercontent.com/191335/168967282-de57c82d-14df-442c-b400-6896ca9d1ce4.png)

I can't reproduce.

Please link a multiplayer *room* with this issue. Closing for now as I dont' beleive this is an issue (maybe someone trolling you).

**Comment by hburn7** [comment]
There may be a miscommunication on the issue. The problem is that if a link exists in the osu! multiplayer room title, an invite to the room will only transport the player into the lobby upon clicking if the player clicks on a portion of the invite that does not have a URL within it. 

From what I tested today, the following should be reproducible. Say "My Title https://www.google.com/" is the title of the lobby. When inviting a player, clicking on the portion of the invite that says "My Title" would load them into the lobby. Clicking on the "google" portion would open Google in a browser. The desired result is to join the lobby, but this wasn't the case in my experience.

This is what I concluded from my testing - if it would be more helpful to provide a video of what happens I am happy to do so.

Edit: Apologies for not being able to link a room right now but that is something I can do as a next step if further investigation is needed.

**Comment by peppy** [comment]
Yes, that is precisely what I was testing, but could not reproduce the issue.

Please provide more details on exactly how to reproduce, and video/screenshots (drag into github, rather than linking to imgur etc.).

**Comment by hburn7** [comment]
I was able to reproduce on `osu!Stable b20220424`.

Steps to reproduce:
- Create a tournament lobby via `!mp make` OR a standard multiplayer lobby via `Multiplayer > New Game` with a valid URL somewhere in the lobby name.
- Receive an invite from someone else (e.g. `!mp invite`)

Video demonstration: https://streamable.com/nzogw0
Multiplayer lobby: https://osu.ppy.sh/mp/100712454

**Comment by peppy** [comment]
Thanks for the extra context. I'll keep this open, but the scenario is a bit complex (and rare) so I'm not sure we will get to resolving this on stable.

I can confirm that this linkify logic is a lot better in [lazer](https://github.com/ppy/osu#running-osu), and the provided example cannot fail there.
